internal/pkg/db/db_test.go

```diff
@@ -93,6 +106,114 @@ func (dbs *dbSuite) TestGetDocumentByID() {
+	}
+
+	spdxDoc, err := dbs.backend.GetDocumentByIDOrAlias("spdx")
+	if err != nil {
+		dbs.Fail("failed retrieving document", "alias", "spdx")
+	}
+
+	dbs.Require().Equal(cdxDoc.GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().Equal(spdxDoc.GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+}
+
+func (dbs *dbSuite) TestGetDocumentsByIDOrAlias() {
+	docs, err := dbs.backend.GetDocumentsByIDOrAlias("cdx", "spdx")
+	if err != nil {
+		dbs.Fail("failed retrieving document", "aliases", "cdx, spdx")
+	}
+
+	dbs.Require().Equal(docs[0].GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().Equal(docs[1].GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+}
+
+func (dbs *dbSuite) TestGetDocumentTags() {
+	tags, err := dbs.backend.GetDocumentTags(dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().NoError(err)
+	dbs.Require().EqualValues([]string{"tag1", "tag2"}, tags)
+}
+
+func (dbs *dbSuite) TestFilterDocumentsByTag() {
+	docs, err := dbs.backend.GetDocumentsByID()
+	dbs.Require().NoError(err)
+
+	// Normal (1 tag, 1 doc)
+	filteredDocs1, err := dbs.backend.FilterDocumentsByTag(docs, "tag1")
+	dbs.Require().NoError(err)
+	dbs.Require().Len(filteredDocs1, 1)
+	dbs.Require().Equal(filteredDocs1[0].GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+
+	// Normal (1 tag, 2 docs)
+	filteredDocs2, err := dbs.backend.FilterDocumentsByTag(docs, "tag2")
+	dbs.Require().NoError(err)
+	dbs.Require().Len(filteredDocs2, 2)
+	dbs.Require().Equal(filteredDocs2[0].GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().Equal(filteredDocs2[1].GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+
+	// Normal (another tag, 1 doc)
+	filteredDocs3, err := dbs.backend.FilterDocumentsByTag(docs, "tag3")
+	dbs.Require().NoError(err)
+	dbs.Require().Len(filteredDocs3, 1)
+	dbs.Require().Equal(filteredDocs3[0].GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+
+	// Normal (multiple tags)
+	filteredDocs4, err := dbs.backend.FilterDocumentsByTag(docs, "tag1", "tag2", "tag3")
+	dbs.Require().NoError(err)
+	dbs.Require().Len(filteredDocs4, 2)
+	dbs.Require().Equal(filteredDocs4[0].GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().Equal(filteredDocs4[1].GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+
+	// Unknown tag
+	filteredDocs5, err := dbs.backend.FilterDocumentsByTag(docs, "unknown_tag")
+	dbs.Require().NoError(err)
+	dbs.Require().Empty(filteredDocs5)
+
+	// No tags
+	filteredDocs6, err := dbs.backend.FilterDocumentsByTag(docs)
+	dbs.Require().NoError(err)
+	dbs.Require().Len(filteredDocs6, 2)
+	dbs.Require().Equal(filteredDocs6[0].GetMetadata().GetId(), dbs.documents[0].GetMetadata().GetId())
+	dbs.Require().Equal(filteredDocs6[1].GetMetadata().GetId(), dbs.documents[1].GetMetadata().GetId())
+}
+
+func (dbs *dbSuite) TestSetAlias() {
```

Comment:
This should be a loop and create a subtest for each test case. See previous example